src/app/services/route-parts/route-parts.service.ts

```diff
@@ -39,6 +39,15 @@ export class RoutePartsService {
       }
 
       const { title, breadcrumb, disabled } = child.snapshot.data;
+
+      if (title === 'Datasets') {
```

Comment:
This is pretty bad. You are hardcoding specific strings in a service that should not know anything about datasets or storage.